Sources/SKCore/CompilationDatabase.swift

```diff
@@ -185,12 +211,22 @@ extension JSONCompilationDatabase: Codable {
 }
 
 extension JSONCompilationDatabase {
-  public init(directory: AbsolutePath, _ fileSystem: FileSystem = localFileSystem) throws {
+  /// Loads the compilation database located in `directory`, if any.
+  /// returns: `nil` if `compile_commands.json` was not found
+  public init?(directory: AbsolutePath, _ fileSystem: FileSystem = localFileSystem) throws {
     let path = directory.appending(component: "compile_commands.json")
-    try self.init(file: path, fileSystem)
+    guard let database = try Self(file: path, fileSystem) else {
+      return nil
+    }
+    self = database
```

Comment:
Also here, couldn’t this be

```suggestion
    try self.init(file: path, fileSystem)
```